runtime/trampolines.m

```diff
@@ -591,3 +613,578 @@
+id xamarin_cgvector_to_nsvalue               (MonoObject *value, guint32 *exception_gchandle) { return [NSValue valueWithCGVector:            *(CGVector *)               mono_object_unbox (value)]; }
+#endif
+#if HAVE_COREANIMATION
+id xamarin_catransform3d_to_nsvalue          (MonoObject *value, guint32 *exception_gchandle) { return [NSValue valueWithCATransform3D:       *(CATransform3D *)          mono_object_unbox (value)]; }
+#endif
+#if HAVE_MAPKIT // Yep, this is defined in MapKit.
+id xamarin_cllocationcoordinate2d_to_nsvalue (MonoObject *value, guint32 *exception_gchandle) { return [NSValue valueWithMKCoordinate:        *(CLLocationCoordinate2D *) mono_object_unbox (value)]; }
+#endif
+#if HAVE_COREMEDIA
+id xamarin_cmtime_to_nsvalue                 (MonoObject *value, guint32 *exception_gchandle) { return [NSValue valueWithCMTime:              *(CMTime *)                 mono_object_unbox (value)]; }
+id xamarin_cmtimemapping_to_nsvalue          (MonoObject *value, guint32 *exception_gchandle) { return [NSValue valueWithCMTimeMapping:       *(CMTimeMapping *)          mono_object_unbox (value)]; }
+id xamarin_cmtimerange_to_nsvalue            (MonoObject *value, guint32 *exception_gchandle) { return [NSValue valueWithCMTimeRange:         *(CMTimeRange *)            mono_object_unbox (value)]; }
+#endif
+#if HAVE_MAPKIT
+id xamarin_mkcoordinatespan_to_nsvalue       (MonoObject *value, guint32 *exception_gchandle) { return [NSValue valueWithMKCoordinateSpan:    *(MKCoordinateSpan *)       mono_object_unbox (value)]; }
+#endif
+id xamarin_scnmatrix4_to_nsvalue             (MonoObject *value, guint32 *exception_gchandle) { return [NSValue valueWithSCNMatrix4:          *(SCNMatrix4 *)             mono_object_unbox (value)]; }
+id xamarin_scnvector3_to_nsvalue             (MonoObject *value, guint32 *exception_gchandle) { return [NSValue valueWithSCNVector3:          *(SCNVector3 *)             mono_object_unbox (value)]; }
+id xamarin_scnvector4_to_nsvalue             (MonoObject *value, guint32 *exception_gchandle) { return [NSValue valueWithSCNVector4:          *(SCNVector4 *)             mono_object_unbox (value)]; }
+#if HAVE_UIKIT
+id xamarin_uiedgeinsets_to_nsvalue           (MonoObject *value, guint32 *exception_gchandle) { return [NSValue valueWithUIEdgeInsets:        *(UIEdgeInsets *)           mono_object_unbox (value)]; }
+id xamarin_uioffset_to_nsvalue               (MonoObject *value, guint32 *exception_gchandle) { return [NSValue valueWithUIOffset:            *(UIOffset *)               mono_object_unbox (value)]; }
+#endif
+
+static void *
+xamarin_get_nsnumber_converter (MonoClass *managedType, MonoMethod *method, bool to_managed, guint32 *exception_gchandle)
+{
+	void * func = NULL;
+	char *fullname = xamarin_class_get_full_name (managedType, exception_gchandle);
+	if (*exception_gchandle != 0)
+		goto exception_handling;
+
+	if (!strcmp (fullname, "System.SByte")) {
```

Comment:
can't we access the native equivalent of TypeCode here ? instead of chaining string comparisons one after the other